syft/messaging/message.py

```diff
@@ -210,10 +210,12 @@ class ObjectMessage(Message):
     to do so.
     """
 
-    def __init__(self, object_):
+    def __init__(self, object_, sender=None, origin_id=None):
         """Initialize the message."""
 
         self.object = object_
+        self.object.sender = sender
+        self.object.origin_id = origin_id
```

Comment:
Actually, on reading further, I think putting these properties on the object makes a lot of sense, but it might make sense to do sooner in the method call chain to avoid having to pass a flag around to decide whether or not to set them.